pkg/scheduler/framework/v1alpha1/types.go

```diff
@@ -549,8 +549,14 @@ func (n *NodeInfo) resetSlicesIfEmpty() {
 func calculateResource(pod *v1.Pod) (res Resource, non0CPU int64, non0Mem int64) {
 	resPtr := &res
 	for _, c := range pod.Spec.Containers {
-		resPtr.Add(c.Resources.Requests)
-		non0CPUReq, non0MemReq := schedutil.GetNonzeroRequests(&c.Resources.Requests)
+		var non0CPUReq, non0MemReq int64
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			resPtr.Add(c.ResourcesAllocated)
```

Comment:
will the ResourcesAllocate field be properly set when upgrading from a cluster that doesn't have this feature enabled (or doesn't have this field at all) to one that does? I assume when api server reads from etcd, it will convert the pod to v1 and at that the conversion logic should probably handle this.

/cc @liggitt

Reply:
yes, as long as the scheduler is speaking to an API server that is aware of the field and has the conversion logic, anything done in conversion will apply to pods that were created in an earlier version when they are read from etcd